src/irmin-http/unix/irmin_http_unix.mli

```diff
@@ -14,6 +14,18 @@
  * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  *)
 
+module type Sock = sig
+  val sock : string
+end
+
+module type Http_client = sig
+  include module type of Cohttp_lwt_unix.Client
+
+  val ctx : unit -> ctx option
+end
+
+module Http_client (P : Sock) : Http_client
```

Comment:
I tried to avoid introducing anything new / making code changes, but I added this to let us use in tests.

Reply:
I wouldn’t worry too much about irmin-http since it will likely be removed soon.

Reply:
Hopefully these changes will make the removal that much easier :smile: